### ISSUE TITLE
pyvips icc_transform: no input profile

### ISSUE THREAD

**Issue by jindili** [jcupitt/libvips#971]
full log output:
```
pyvips.error.Error: unable to call icc_transform
icc_transform: no input profile
```

pyvips code:
```
im = pyvips.Image.new_from_file('path/to/some/image')
... after many processing ...
cmyk = im.icc_transform('rgb2cmyk.icc', 
                        embedded=True,
                        input_profile='sRGB.icm')
cmyk.tiffsave("out.tif", compression="lzw", predictor="horizontal")
```
the sRGB.icm for input_profile is the file come with nip2
my vips version: vips-8.6.3-Thu Mar  8 15:18:35 UTC 2018

I use this code to processing quite lot of images, on some of them this error occur.

but if I save the image after many processing to disk and load it back then icc_transform it works fine.

**Comment by jcupitt** [comment]
Hello @jindili, perhaps you are running out of file descriptors? It might explain why lcms is unable to load the sRGB profile. 

You could try raising the number of file descriptors, or perhaps forcing a Python GC for each image you process.

**Comment by jindili** [comment]
Hi @jcupitt 

How can I raising the number of file descriptors in pyvips, 
Searching out the pyvips online documents at
https://jcupitt.github.io/pyvips/
but could not find something related.

**Comment by jindili** [comment]
I try to locate the error , here is the snippet which can reproduce the error:

```
import pyvips

width = 1000

img = pyvips.Image.black(width, width)
img = img.bandjoin(0).bandjoin(0).bandjoin(0)
cmyk = img.icc_transform("cmyk.icm", embedded=False, input_profile="sRGB.icm")
cmyk.tiffsave("cmyk.tif", compression="lzw", predictor="horizontal")
```

which always produce error:

```
pyvips.error.Error: unable to call icc_transform
icc_transform: no input profile
```

if use this code
```
cmyk = img.icc_export(output_profile = "cmyk.icm")
```

output:
```
before icc: <pyvips.Image 1000x1000 uchar, 4 bands, b-w>
after icc: <pyvips.Image 1000x1000 uchar, 7 bands, cmyk>
```
got 7 bands cmyk image, which is unwanted.

**Comment by jcupitt** [comment]
Hello @jindili,

Thank you for looking into this. I poked about inside and the problem (as you say) comes from the input image not being tagged as RGB. When `icc_transform` tries to verify that the input profile matches the image, it fails with that misleading error message.

I've added something to improve the way the image colourspace is guessed in this case. This program now works fine:

```python
import pyvips
  
width = 1000

img = pyvips.Image.black(width, width)
img = img.bandjoin(0).bandjoin(0)
cmyk = img.icc_transform("cmyk.icm", embedded=False, input_profile="sRGB.icm")
cmyk.tiffsave("cmyk.tif", compression="lzw", predictor="horizontal")
```

Thanks for reporting this!
